delivery_ups_oca/models/ups_request.py

```diff
@@ -111,17 +112,27 @@ def _label_data(self):
 
     def _prepare_create_shipping(self, picking):
         """Return a dict that can be passed to the shipping endpoint of the UPS API"""
-        if picking.package_ids:
+        if self.use_packages_from_picking and picking.package_ids:
             packages = [
                 self._quant_package_data_from_picking(package, picking, True)
                 for package in picking.package_ids
             ]
         else:
-            packages = [
-                self._quant_package_data_from_picking(
-                    self.default_packaging_id, picking, False
-                )
-            ]
+            packages = []
+            package_info = self._quant_package_data_from_picking(
+                self.default_packaging_id, picking, False
+            )
+            package_weight = round(
+                (picking.shipping_weight / picking.number_of_packages), 2
+            )
+            for i in range(0, picking.number_of_packages):
+                package_item = package_info
+                package_name = "%s (%s)" % (picking.name, i + 1)
+                package_item["Description"] = package_name
+                package_item["NumOfPieces"] = "1"
```

Comment:
NumOfPieces? 1

Reply:
This value is not too important (it is only needed in some countries where it is mandatory) and it is intended to know how many elements this package contains (if odoo packages are used, this value is defined correctly).